dev_tools/run_doctest.py

```diff
@@ -0,0 +1,233 @@
+    if include_modules:
+        import cirq
+        import numpy
+        import sympy
+        import pandas
+        base_globals = {'cirq': cirq, 'np': numpy, 'sympy': sympy, 'pd': pandas}
+    else:
+        base_globals = {}
+
+    try_print('Loading tests   ', end='')
+
+    def make_test(file_path: str) -> Doctest:
+        try_print('.', end='', flush=True)
+        mod = import_file(file_path)
+        glob = make_globals(mod)
+        return Doctest(file_path, mod, glob)
+
+    def make_globals(mod: ModuleType) -> Dict[str, Any]:
+        if include_local:
+            glob = dict(mod.__dict__)
+            glob.update(base_globals)
+            return glob
+        else:
+            return dict(base_globals)
+
+    tests = [make_test(file_path) for file_path in file_paths]
+    try_print()
+    return tests
+
+
+def exec_tests(tests: Iterable[Doctest],
+               quiet: bool = True) -> Tuple[doctest.TestResults, List[str]]:
+    """Runs a list of `Doctest`s and collects and returns any error messages.
+
+    Args:
+        tests: The tests to run
+
+    Returns: A tuple containing the results (# failures, # attempts) and a list
+        of the error outputs from each failing test.
+    """
+    if not quiet:
+        try_print = print
+    else:
+        try_print = lambda *args, **kwargs: None
+    try_print('Executing tests ', end='')
+
+    failed, attempted = 0, 0
+    error_messages = []
+    for test in tests:
+        out = OutputCapture()
+        with out:
+            r = test.run()
+        failed += r.failed
+        attempted += r.attempted
+        if r.failed != 0:
+            try_print('F', end='', flush=True)
+            error = shell_tools.highlight(
+                '{}\n{} failed, {} passed, {} total\n'.format(
+                    test.file_name, r.failed, r.attempted - r.failed,
+                    r.attempted), shell_tools.RED)
+            error += out.content()
+            error_messages.append(error)
+        else:
+            try_print('.', end='', flush=True)
+
+    try_print()
+
+    return doctest.TestResults(failed=failed,
+                               attempted=attempted), error_messages
+
+
+def import_file(file_path: str) -> ModuleType:
+    """Finds and runs a python file as if it were imported with an `import`
+    statement.
+
+    Args:
+        file_path: The file to import.
+
+    Returns:
+        The imported module.
+    """
+    mod_name = os.path.basename(file_path)
+    if mod_name.endswith('.py'):
+        mod_name = mod_name[:-3]
+    # Find and create the module
+    spec = importlib.util.spec_from_file_location(mod_name, file_path)
+    mod = importlib.util.module_from_spec(spec)
+    # Run the code in the module (but not with __name__ == '__main__')
+    spec.loader.exec_module(mod)  # type: ignore
+    return mod
+
+
+def main():
+    quiet = len(sys.argv) >= 2 and sys.argv[1] == '-q'
+
+    file_names = glob.glob('cirq/**/*.py', recursive=True)
```

Comment:
This should be `openfermioncirq/**/*.py`.

Reply:
Done